### ISSUE TITLE
`writesSharedRun` does not know that `declaration-colon-newline-after` moves its question behind a comment the space rule's write puts on the colon's line, and refuses a write that would have satisfied both rules

### ISSUE THREAD

**Issue by firefoxic** [stylelint-stylistic/stylelint-stylistic#590]
### What minimal example or steps are needed to reproduce the bug?

```css
a { b:
/*c*/
x; }
```

The two colon rules read the run behind the colon as one run they settle between them: the space rule asks `writesSharedRun` whether the newline rule, listed behind it, would be content with a space there. The model answers as if the newline rule always asked about the run behind the colon. It does not: where a block comment stands on the colon's own line, `declaration-colon-newline-after` asks about the run behind that comment (#400), and writing a space over the break here is exactly what puts the comment on the colon's line. The file the space rule would write, `a { b: /*c*/\nx; }`, is clean under both rules; the model calls the pair a contradiction, the space rule reports and writes nothing, and the warning stands on every run of `--fix`.

### What minimal configuration is needed to reproduce the bug?

```json
{
	"plugins": [
		"@stylistic/stylelint-plugin"
	],
	"rules": {
		"@stylistic/declaration-colon-space-after": "always",
		"@stylistic/declaration-colon-newline-after": "always-multi-line"
	}
}
```

Either order of the two, and `never` in place of `always` — four configurations in all.

### What did you expect to happen?

`--fix` writes `a { b: /*c*/\nx; }` and nothing is reported afterwards, since that file answers both rules.

### What actually happened?

`--fix` leaves the file as it is and reports `Expected single space after ":"` on every run; under `never`, `Unexpected whitespace after ":"`.

### Notes

Found by the race of every colon-rule pair over the corpus of the sweep `colon-lineness` while closing #389, which is what makes the shape reachable: before that branch the value `x` counted as single-line, `always-multi-line` was silent, and the space rule wrote its space unasked. Eight texts of that corpus show it, each under the four configurations above: the comment with a break inside or none, standing at the head of a worded value on a plain or a custom property, and — with a break inside, or none with a break behind it — alone as the whole value of a custom property, every one with a break behind the colon, and all but two with a break behind the comment as well (`a { b:\n/*c*/\nx; }` for one, and `a { --b:\n/*c\n*/ ; }` for one of the two). No other pair of the four rules about these runs puts a row of this kind on the board.

The shape the model lacks is the one `sharedRunsOf` already knows for a comment that stands on the colon's line before any write: there the newline rule reads the run behind the comment and is no reader of the head run. What is missing is asking that question of the run as the write would leave it rather than as it stands.
